Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/Home.kt

```diff
@@ -367,34 +397,54 @@ fun HomeCategoryTabIndicator(
     )
 }
 
+private val FEATURED_PODCAST_IMAGE_WIDTH_DP = 160.dp
+private val FEATURED_PODCAST_IMAGE_HEIGHT_DP = 180.dp
+
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun FollowedPodcasts(
+    pagerState: PagerState,
     items: PersistentList<PodcastWithExtraInfo>,
     modifier: Modifier = Modifier,
     onPodcastUnfollowed: (String) -> Unit,
 ) {
-    // TODO: Update this component to a carousel once better support is available
-    val lastIndex = items.size - 1
-    LazyRow(
-        modifier = modifier,
+    val coroutineScope = rememberCoroutineScope()
+
+    var horizontalPadding by remember { mutableStateOf(0.dp) }
+    val density = LocalDensity.current
+    val screenWidth = LocalConfiguration.current.screenWidthDp.dp
+    HorizontalPager(
+        state = pagerState,
+        modifier = modifier.onSizeChanged { size ->
+            // TODO: this is not quite performant since it requires 2 passes to compute the content
+            // padding. This should be revisited once a carousel component is available.
+            // Alternatively, version 1.7.0-alpha05 of Compose Foundation supports `snapPosition`
+            // which solves this problem and avoids this calculation altogether. Once 1.7.0 is
+            // stable, this implementation can be updated.
+            horizontalPadding = with(density) {
+                (size.width.toDp() - FEATURED_PODCAST_IMAGE_WIDTH_DP) / 2
+            }
```

Comment:
`BoxWithConstraints` might be more desirable here because this two frame padding calculation is visible to the user. It seems weird to me you have to do this in the first place though, I would have thought you could do some combination of pageSpacing, custom page size and horizontal padding to get it to centre

Reply:
Thanks - will change to `BoxWithConstraints`. I was also hoping to accomplish the same effect without needing the parent's size but unfortunately, that's needed. `snapPolicy` is really what we want here but that's still in alpha.